cmd/generic-handlers.go

```diff
@@ -337,15 +341,11 @@ func setReservedBucketHandler(h http.Handler) http.Handler {
 }
 
 func (h minioReservedBucketHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	switch {
-	case guessIsRPCReq(r), guessIsBrowserReq(r), guessIsHealthCheckReq(r), guessIsMetricsReq(r), isAdminReq(r):
-		// Allow access to reserved buckets
-	default:
-		// For all other requests reject access to reserved buckets
-		bucketName, _ := request2BucketObjectName(r)
-		if isMinioReservedBucket(bucketName) || isMinioMetaBucket(bucketName) {
-			browser := guessIsBrowserReq(r)
-			writeErrorResponse(r.Context(), w, errorCodes.ToAPIErr(ErrAllAccessDisabled), r.URL, browser)
+	// For all other requests reject access to reserved buckets
+	bucketName, _ := request2BucketObjectName(r)
+	if isMinioReservedBucket(bucketName) || isMinioMetaBucket(bucketName) {
+		if !guessIsRPCReq(r) && !guessIsBrowserReq(r) && !guessIsHealthCheckReq(r) && !guessIsMetricsReq(r) && !isAdminReq(r) {
```

Comment:
This changes behavior. These will now only get triggered on reserved or meta bucket.

I would also expect these checks to be cheaper than extracting the bucket name.

What is the background for this change? It seems to me that keeping it the way it is is better.